src/types.ts

```diff
@@ -0,0 +1,81 @@
+export interface SophosConfig {
+  clientId: string
+  clientSecret: string
+  partnerId?: string
+}
+
+export interface SophosTenant {
+  id?: string
+  name?: string
+  dataGeography?: string
+  dataRegion?: string
+  billingType?: string
+  partner?: object
+  organization?: string
+  apiHost?: string
+}
+
+export interface SophosEndpoint {
+  id?: string
+  type?: string
+  tenant?: {
+    id?: string
+  }
+  hostname?: string
+  health?: {
+    overall?: string
+    threats?: {
+      status?: string
+    }
+    services?: {
+      status?: string
+      serviceDetails?: SophosEndpointService[]
+    }
+  }
+  os?: {
+    isServer?: boolean
+    platform?: string
+    name?: string
+    majorVersion?: number
+    minorVersion?: number
+    build?: number
+  }
+  ipv4Addresses?: string[]
+  macAddresses?: string[]
+  associatedPerson?: {
+    name?: string
+    viaLogin?: string
+    id?: string
+  }
+  tamperProtectionEnabled?: boolean
+  assignedProducts?: SophosEndpointServiceAssignedProduct[]
+  lastSeenAt?: string
+  encryption?: {
+    volumes?: SophosEndpointEncryptionVolume[]
+  }
+}
+
+export interface SophosEndpointService {
+  name?: string
+  status?: string
+}
+
+export interface SophosEndpointServiceAssignedProduct {
+  code?: string
+  version?: string
+  status?: string
+}
+export interface SophosEndpointEncryptionVolume {
+  volumeId?: string
+  status?: string
+}
+
+export interface SophosAccessToken {
+  access_token?: string
+  errorCode?: string
+  expires_in?: string
+  message?: string
+  refresh_token?: string
+  token_type?: string
+  trackingId?: string
+}
```

Comment:
I'll ask @noticeeverything to weigh in on this one, but I think we should remove the `Sophos` prefix from the interface names to make it more idiomatic with how NPM packages typically include types.

Additionally, it looks like there are a fair amount of optional parameters. Have you encountered situations where they are optional (e.g., `id` on the `Tenant` interface)? Using these interfaces in practice would likely lead to unnecessary optional chaining and nullish coalescing.

Thanks for putting together the types, going to be super useful!

Reply:
I think the prefix is quite often very helpful to have, but not always. For instance, if the Sophos lib exported a `Device` and Addigy did the same, and I wanted to reference both types in my implementation, I'd have to do something like `import { Device as SophosDevice } from 'sophos'`, which is kinda inelegant, but sometimes unavoidable. Probably the deciding factor is how generic a type is or not. I could envision `Endpoint` being a generic type and `SophosEndpoint` extending it, which would make the prefix useful for anything consuming the Sophos lib. On the other hand, something like an `EndpointEncryptionVolume` might be more specific to Sophos and unlikely to cause naming conflicts for consumers.

As for optional properties, I would absolutely avoid them if possible. It's often difficult to avoid, however. But in a case like the `SophosAccessToken` I gotta imagine that the `access_token` property at least can never be undefined. I'd look for any cases like that and make the property required.